agr_literature_service/lit_processing/update_pubmed_papers.py

```diff
@@ -1029,72 +1057,90 @@ def get_orcid_data(db_session):
     return orcid_dict
 
 
-def get_author_data(db_session, mod, reference_id_list):
+def get_author_data(db_connection, mod, reference_id_list):
 
     reference_id_to_authors = {}
 
-    allAuthors = None
     if mod and len(reference_id_list) > query_cutoff:
-        allAuthors = db_session.query(AuthorModel).join(ReferenceModel.author).outerjoin(ReferenceModel.mod_corpus_association).outerjoin(ModCorpusAssociationModel.mod).filter(ModModel.abbreviation == mod).order_by(AuthorModel.reference_id, AuthorModel.order).all()
+        author_limit = 500000
+        for index in range(500):
+            offset = index * author_limit
+            rs = db_connection.execute("select a.reference_id, a.orcid, a.first_author, a.order, a.corresponding_author, a.name, a.affiliations, a.first_name, a.last_name from author a, mod_corpus_association mca, mod m where a.reference_id = mca.reference_id and mca.mod_id = m.mod_id and m.abbreviation = '" + mod + "' order by a.reference_id, a.order limit " + str(author_limit) + " offset " + str(offset))
+            rows = rs.fetchall()
+            if len(rows) == 0:
+                break
+            for x in rows:
+                authors = []
+                reference_id = x[0]
+                if reference_id in reference_id_to_authors:
+                    authors = reference_id_to_authors[reference_id]
+                authors.append({"orcid": x[1],
+                                "first_author": x[2],
+                                "order": x[3],
+                                "corresponding_author": x[4],
+                                "name": x[5],
+                                "affiliations": x[6] if x[6] else [],
+                                "first_name": x[7],
+                                "last_name": x[8]})
+                reference_id_to_authors[reference_id] = authors
     elif reference_id_list and len(reference_id_list) > 0:
-        allAuthors = db_session.query(AuthorModel).filter(AuthorModel.reference_id.in_(reference_id_list)).all()
-
-    if allAuthors is None:
-        return reference_id_to_authors
-
-    for x in allAuthors:
-        authors = []
-        if x.reference_id in reference_id_to_authors:
-            authors = reference_id_to_authors[x.reference_id]
-        authors.append(x)
-        reference_id_to_authors[x.reference_id] = authors
-
+        # name & order are keywords in postgres so have use alias 'a' for table name
+        ref_ids = ", ".join([str(x) for x in reference_id_list])
+        raw_sql = "SELECT a.reference_id, a.orcid, a.first_author, a.order, a.corresponding_author, a.name, a.affiliations, a.first_name, a.last_name FROM author a WHERE reference_id IN (" + ref_ids + ") order by a.reference_id, a.order"
+        rs = db_connection.execute(raw_sql)
+        rows = rs.fetchall()
+        for x in rows:
```

Comment:
This code seems to be duplicated. Can you create a function to avoid duplication?

Reply:
good idea

Reply:
Fixed.

Reply:
looks good, thanks @sweng66